src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/osgi/OsgiFramework.kt

```diff
@@ -132,6 +133,55 @@ class OsgiFramework(sync: InstanceSync) : InstanceService(sync) {
         enableComponent(pid)
     }
 
+    fun findConfiguration(pid: String): Configuration? {
+        return try {
+            sync.http.get("$CONFIGURATION_PATH/$pid?post=true") { asObjectFromJson(it, Configuration::class.java) }
+        } catch (e: AemException) {
+            null
+        }
+    }
+
+    fun getConfiguration(pid: String): Configuration {
+        return findConfiguration(pid)
+                ?: throw InstanceException("OSGi configuration for PID $pid cannot be found on $instance.")
+    }
+
+    fun getConfigurations(pid: String): List<Configuration> {
+        return getConfigurationsList()
+                .filter { pid == it.fpid }
+                .mapNotNull { findConfiguration(it.id) }
+    }
+
+    fun saveConfiguration(pid: String, properties: Map<String, Any>) {
+        val currentConfiguration = getConfiguration(pid)
```

Comment:
just `current` / you are in the scope of `saveConfiguration` (we know that everything is about same configuration here)

Reply:
done